oracle/config/metrics.go

```diff
@@ -2,99 +2,29 @@ package config
 
-	// OracleMetrics is the config for the oracle metrics
-	OracleMetrics OracleMetricsConfig `mapstructure:"oracle_metrics" toml:"oracle_metrics"`
-
-	// AppMetrics is the config for the app metrics
-	AppMetrics AppMetricsConfig `mapstructure:"app_metrics" toml:"app_metrics"`
-}
-
-// ValidateBasic performs basic validation of the config.
-func (c *MetricsConfig) ValidateBasic() error {
-	if (c.OracleMetrics.Enabled || c.AppMetrics.Enabled) && len(c.PrometheusServerAddress) == 0 {
-		return fmt.Errorf("must supply a prometheus server address if metrics are enabled")
-	}
-
-	return c.AppMetrics.ValidateBasic()
-}
-
-// OracleMetricsConfig is the config for the oracle metrics. These are also utililized to enable
-// provider metrics.
-type OracleMetricsConfig struct {
 	// Enabled indicates whether metrics should be enabled.
 	Enabled bool `mapstructure:"enabled" toml:"enabled"`
 }
 
-// AppMetricsConfig is the config for the app metrics. Specifically, this is used to enable validator
-// metrics as proposals and vote extensions are being constructed. Nodes that are not participating
-// in consensus should not enable app metrics.
-type AppMetricsConfig struct {
-	// Enabled indicates whether app side metrics should be enabled.
-	Enabled bool `mapstructure:"enabled" toml:"enabled"`
-
-	// ValidatorConsAddress is the validator's consensus address. Validator's must register their
-	// consensus address in order to enable app side metrics.
-	ValidatorConsAddress string `mapstructure:"validator_cons_address" toml:"validator_cons_address"`
-}
-
 // ValidateBasic performs basic validation of the config.
-func (c *AppMetricsConfig) ValidateBasic() error {
-	if c.Enabled {
-		_, err := sdk.ConsAddressFromBech32(c.ValidatorConsAddress)
-		return err
-	}
-
-	return nil
-}
-
-// ConsAddress returns the validator's consensus address
-func (c *AppMetricsConfig) ConsAddress() (sdk.ConsAddress, error) {
-	if c.Enabled {
-		return sdk.ConsAddressFromBech32(c.ValidatorConsAddress)
-	}
-
-	return nil, nil
-}
-
-// ReadMetricsConfigFromFile reads a config from a file and returns the config.
-func ReadMetricsConfigFromFile(path string) (MetricsConfig, error) {
-	// Read in config file.
-	viper.SetConfigFile(path)
-	viper.SetConfigType("toml")
-
-	if err := viper.ReadInConfig(); err != nil {
-		return MetricsConfig{}, err
-	}
-
-	// Check required fields.
-	requiredFields := []string{"prometheus_server_address", "oracle_metrics", "app_metrics"}
-	for _, field := range requiredFields {
-		if !viper.IsSet(field) {
-			return MetricsConfig{}, fmt.Errorf("required field %s is missing in config", field)
-		}
-	}
-
-	// Unmarshal the config.
-	var config MetricsConfig
-	if err := viper.Unmarshal(&config); err != nil {
-		return MetricsConfig{}, err
+func (c *MetricsConfig) ValidateBasic() error {
+	if !c.Enabled {
+		return nil
 	}
 
-	if err := config.ValidateBasic(); err != nil {
-		return MetricsConfig{}, err
+	if len(c.PrometheusServerAddress) == 0 {
```

Comment:
Can we validate this beyond being empty as well?

Reply:
added the same parseURL functionality you recommended above.